GLTFSDK.Samples/Serialize/Source/main.cpp

```diff
@@ -0,0 +1,294 @@
+        std::experimental::filesystem::path pathFileExt = pathFile.extension();
+
+        auto MakePathExt = [](const std::string& ext)
+        {
+            return "." + ext;
+        };
+
+        std::unique_ptr<ResourceWriter> resourceWriter;
+
+        // If the file has a '.gltf' extension then create a GLTFResourceWriter
+        if (pathFileExt == MakePathExt(GLTF_EXTENSION))
+        {
+            resourceWriter = std::make_unique<GLTFResourceWriter>(std::move(streamWriter));
+        }
+
+        // If the file has a '.glb' extension then create a GLBResourceWriter. This class derives
+        // from GLTFResourceWriter and adds support for writing manifests to a GLB container's
+        // JSON chunk and resource data to the binary chunk.
+        if (pathFileExt == MakePathExt(GLB_EXTENSION))
+        {
+            resourceWriter = std::make_unique<GLBResourceWriter>(std::move(streamWriter));
+        }
+
+        if (!resourceWriter)
+        {
+            throw std::runtime_error("Command line argument path filename extension must be .gltf or .glb");
+        }
+
+        // The Document instance represents the glTF JSON manifest
+        Document document;
+
+        std::string accessorIdIndices;
+        std::string accessorIdPositions;
+
+        // Use the BufferBuilder helper class to simplify the process of
+        // constructing valid glTF Buffer, BufferView and Accessor entities
+        BufferBuilder bufferBuilder(std::move(resourceWriter));
+
+        CreateTriangleResources(document, bufferBuilder, accessorIdIndices, accessorIdPositions);
+        CreateTriangleEntities(document, accessorIdIndices, accessorIdPositions);
+
+        std::string manifest;
+
+        try
+        {
+            // Serialize the glTF Document into a JSON manifest
+            manifest = Serialize(document, SerializeFlags::Pretty);
+        }
+        catch (const GLTFException& ex)
+        {
+            std::stringstream ss;
+
+            ss << "Microsoft::glTF::Serialize failed: ";
+            ss << ex.what();
+
+            throw std::runtime_error(ss.str());
+        }
+
+        auto& gltfResourceWriter = bufferBuilder.GetResourceWriter();
+
+        if (auto glbResourceWriter = dynamic_cast<GLBResourceWriter*>(&gltfResourceWriter))
+        {
+            glbResourceWriter->Flush(manifest, pathFile.u8string()); // A GLB container isn't created until the GLBResourceWriter::Flush member function is called
+        }
+        else
+        {
+            gltfResourceWriter.WriteExternal(pathFile.u8string(), manifest); // Binary resources have already been written, just need to write the manifest
+        }
+    }
+}
+
+#if defined _WIN32 && defined _UNICODE
+int wmain(int argc, wchar_t* argv[])
+#else // _WIN32 & _UNICODE
+int main(int argc, char* argv[])
+#endif
+{
+    try
+    {
+        if (argc != 2U)
+        {
+            throw std::runtime_error("Unexpected number of command line arguments");
```

Comment:
```suggestion
            throw std::runtime_error("Unexpected number of command line arguments\n");
```

Reply:
I've added a trailing newline in the catch block when the error message is output to std::cerr